test/new_relic/agent/instrumentation/rails/action_mailbox_subscriber.rb

```diff
@@ -0,0 +1,151 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require_relative '../../../../test_helper'
+require 'new_relic/agent/instrumentation/action_mailer_subscriber'
+
+module NewRelic::Agent::Instrumentation
+  class TestMailbox < ActionMailbox::Base
+    def process; end
+  end
+
+  class InboundEmail
+    def bounced?; false; end
+    def delivered!; end
+    def failed!; end
+    def instrumentation_payload; end
+    def processing!; end
+  end
+
+  class ActionMailboxSubscriberTest < Minitest::Test
+    ACTION = 'process'
+    NAME = "#{ACTION}.action_mailbox"
+    ID = 1946
+    SUBSCRIBER = NewRelic::Agent::Instrumentation::ActionMailboxSubscriber.new
+    MAILBOX = TestMailbox.new(InboundEmail.new)
+
+    def test_start
+      in_transaction do |txn|
+        time = Time.now.to_f
+        SUBSCRIBER.start(NAME, ID, {mailbox: MAILBOX})
+        segment = txn.segments.last
+
+        assert_in_delta time, segment.start_time
+        assert_equal "Ruby/ActionMailbox/#{MAILBOX.class.name}/#{ACTION}", segment.name
+      end
+    end
+
+    def test_start_when_not_traced
+      SUBSCRIBER.state.stub :is_execution_traced?, false do
+        in_transaction do |txn|
+          SUBSCRIBER.start(NAME, ID, {})
+
+          assert_empty txn.segments
+        end
+      end
```

Comment:
This test was very helpful to me for the Active Support ticket! Thanks for writing it!